### PR TITLE
SI-3995 Exclude companions with an existential prefix.

### PR DESCRIPTION
**Pull request by retronym**
In `(qual: Q).apply(expr)` where `expr` must be implictily
converted to a path dependent type `T` defined in `qual`,
we were looking for companion implicits via a path prefixed
by an existential skolem `_1`. These aren't much good to us,
as when we try to feed them into `mkAttributedQualifer`, a
crash rightly ensues.

This commit excludes companions prefixed by an existentially
bound path.

Review by @adriaanm

### CODE REVIEW

File: test/files/neg/t3995.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  // fails trying to mkAttributedQualifier for pre = Skolem(_1 <: Lift with Singletom).F0
+  // should this even have shown up in `companionImplicitMap`? It says that:
+  //
+  // "@return For those parts that refer to classes with companion objects that
+  // can be accessed with unambiguous stable prefixes, the implicits infos
+  // which are members of these companion objects."
+  //
+  // The skolem is stable, but it doen't seem much good to us
+  (new Lift).apply("")
```
Comment (xeno-by) [inline test/files/neg/t3995.scala:31]: Would `l.apply("")` work?

Comment (retronym) [inline test/files/neg/t3995.scala:31]: Yeah, that's still fine. I'll add that to the test case.

Comment (xeno-by) [inline test/files/neg/t3995.scala:31]: Crashes for me in unpatched 2.10.x

Comment (retronym) [inline test/files/neg/t3995.scala:31]: Odd, can you post your full test? I've just pushed an expanded version of this test which allows `l.apply("")` and expands it to `Test.this.l.apply(Test.this.l.F0.f2f0(""));`

Comment (xeno-by) [inline test/files/neg/t3995.scala:31]: My bad, it crashed with `scala.NotImplementedError: an implementation is missing`.

Comment (retronym) [inline test/files/neg/t3995.scala:31]: Wait, you're trying to _run_ code to test it? :wink:

Comment (xeno-by) [inline test/files/neg/t3995.scala:31]: I can't help myself. REPL is so convenient :)

**Comment by retronym** [comment]
Paging @adriaanm, 

:musical_score: All I want for Christmas is a LGTM...

**Comment by paulp** [comment]
LGTM. Ho ho ho.

**Comment by adriaanm** [comment]
Thanks, Santa, but where's your beard?

I was fretting over how this corresponds to the spec. I have stopped worrying and it started LingGTM when I realized the candidates you exclude here would fail later anyway -- sometimes it helps to look at the bug report. 

Anyway, is there a more general criterion for when we can't stably refer to the qualifier? Could an existential be acceptable in certain cases?

**Comment by paulp** [comment]
That is like word for word what I wrote, and then deleted for the terser "LGTM", after I decided that I'd do better to just go attempt to consolidate the various stability definitions.

**Pull request merged**